pandas/core/dtypes/common.py

```diff
@@ -1786,3 +1786,27 @@ def pandas_dtype(dtype) -> DtypeObj:
         raise TypeError(f"dtype '{dtype}' not understood")
 
     return npdtype
+
+
+def groupby_result_dtype(dtype, how) -> DtypeObj:
```

Comment:
this doesn't belong in here.

Reply:
I moved it to _Groupby._result_dtype as a static method.